ktor-client/ktor-client-core/build.gradle.kts

```diff
@@ -8,6 +8,7 @@ kotlin.sourceSets {
     commonMain {
         dependencies {
             api(project(":ktor-http"))
+            api(project(":ktor-http:ktor-http-cio"))
```

Comment:
this dependency is a bit suspicious :)

May be it will be better to introduce this support in `ktor-http` (or in `ktor-client-core` directly) and just deprecate API in `CIO`?

Reply:
The `ktor-http-cio` stands for ktor HTTP coroutines io. It is reused both in server and client. Could you tell me what problems are with this dependency?

Reply:
Mostly, there are no problems, except for naming confusion between CIO engines and half-related to them `ktor-http-cio`. Currently `ktor-http-cio` contains mostly default implementation for features which are not available in some engines (e.g like this multi-part support). And after this PR it will become a dependency of both `ktor-client-core` and `ktor-server-core`, so it will makes no sense for me to even have `ktor-http-cio` and it will be better to just introduce this functionality/API in `ktor-http`.